### PR TITLE
Add go modules

### PR DESCRIPTION
**Pull request by juliolustosa**
Godep to go modules

### CODE REVIEW

**Comment by localleon** [comment]
Adding Go Modules is a great idea. +1 . This PR also looks very clean and functional

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,17 +3,19 @@ version: 2
...
+    
     steps:
       - checkout
       - run:
           name: "Create a temp directory for artifacts"
           command: |
             mkdir -p /tmp/artifacts
       - run:
+          environment:
+            GOPROXY: "https://proxy.golang.org"
```
[the remark this answers is not in the document]
Comment (juliolustosa) [reply .circleci/config.yml]: ok

File: Makefile [hunk 2]
```diff
@@ -28,15 +27,15 @@ LDFLAGS := '-X "main.version=$(TAG)" -X "main.commit=$(COMMIT)" -X "main.date=$(
 
 ## Builds Go project to the executable file huskyci
 build:
-	cd api && GOOS=linux GOARCH=amd64 $(GO) build -ldflags $(LDFLAGS) -o "$(HUSKYCIBIN)"
+	cd api && GOOS=linux GOARCH=amd64 $(GO) build -mod vendor -ldflags $(LDFLAGS) -o "$(HUSKYCIBIN)"
```
Comment (nettoclaudio) [inline Makefile]: Changing of directory makes vendor local useless. Can you fix that?

File: Makefile [hunk 3]
```diff
@@ -28,15 +27,15 @@ LDFLAGS := '-X "main.version=$(TAG)" -X "main.commit=$(COMMIT)" -X "main.date=$(
...
 
 ## Builds client to the executable file huskyci-client
 build-client:
-	cd client/cmd && $(GO) build -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
+	cd client/cmd && $(GO) build -mod vendor -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
```
Comment (nettoclaudio) [inline Makefile]: Same here...

File: Makefile [hunk 4]
```diff
@@ -28,15 +27,15 @@ LDFLAGS := '-X "main.version=$(TAG)" -X "main.commit=$(COMMIT)" -X "main.date=$(
...
 
 ## Builds client to the executable file huskyci-client
 build-client-linux:
-	cd client/cmd && GOOS=linux GOARCH=amd64 $(GO) build -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
+	cd client/cmd && GOOS=linux GOARCH=amd64 $(GO) build -mod vendor -o "$(HUSKYCICLIENTBIN)" && mv "$(HUSKYCICLIENTBIN)" ../..
```
Comment (nettoclaudio) [inline Makefile]: ... and here.

**Comment by juliolustosa** [comment]
Thank you so much

**Pull request merged**